common/gateway-storage/src/models.rs

```diff
@@ -0,0 +1,163 @@
+    pub derived_aes128_ctr_blake3_hmac_keys_bs58: String,
+}
+
+pub struct StoredMessage {
+    pub id: i64,
+    #[allow(dead_code)]
+    pub client_address_bs58: String,
+    pub content: Vec<u8>,
+}
+
+#[derive(Debug, Clone, FromRow)]
+pub struct PersistedBandwidth {
+    #[allow(dead_code)]
+    pub client_address_bs58: String,
+    pub available: i64,
+    pub freepass_expiration: Option<OffsetDateTime>,
+}
+
+impl From<PersistedBandwidth> for AvailableBandwidth {
+    fn from(value: PersistedBandwidth) -> Self {
+        AvailableBandwidth {
+            bytes: value.available,
+            freepass_expiration: value.freepass_expiration,
+        }
+    }
+}
+
+impl From<Option<PersistedBandwidth>> for AvailableBandwidth {
+    fn from(value: Option<PersistedBandwidth>) -> Self {
+        match value {
+            None => AvailableBandwidth::default(),
+            Some(b) => b.into(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, FromRow)]
+pub struct SpentCredential {
+    #[allow(dead_code)]
+    pub blinded_serial_number_bs58: String,
+    #[allow(dead_code)]
+    pub was_freepass: bool,
+    #[allow(dead_code)]
+    pub client_address_bs58: String,
+}
+
+#[cfg(feature = "wireguard")]
+#[derive(Debug, Clone, FromRow)]
+pub struct WireguardPeer {
+    pub public_key: String,
+    pub preshared_key: Option<String>,
+    pub protocol_version: Option<i64>,
+    pub endpoint: Option<String>,
+    pub last_handshake: Option<sqlx::types::chrono::NaiveDateTime>,
+    pub tx_bytes: i64,
+    pub rx_bytes: i64,
+    pub persistent_keepalive_interval: Option<i64>,
+    pub allowed_ips: Vec<u8>,
+    pub suspended: bool,
+}
+
+#[cfg(feature = "wireguard")]
+impl From<defguard_wireguard_rs::host::Peer> for WireguardPeer {
+    fn from(value: defguard_wireguard_rs::host::Peer) -> Self {
+        WireguardPeer {
+            public_key: value.public_key.to_string(),
+            preshared_key: value.preshared_key.as_ref().map(|k| k.to_string()),
+            protocol_version: value.protocol_version.map(|v| v as i64),
+            endpoint: value.endpoint.map(|e| e.to_string()),
+            last_handshake: value.last_handshake.and_then(|t| {
+                if let Ok(d) = t.duration_since(std::time::UNIX_EPOCH) {
+                    if let Ok(millis) = d.as_millis().try_into() {
+                        sqlx::types::chrono::DateTime::from_timestamp_millis(millis)
+                            .map(|d| d.naive_utc())
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            }),
+            tx_bytes: value.tx_bytes as i64,
+            rx_bytes: value.rx_bytes as i64,
+            persistent_keepalive_interval: value.persistent_keepalive_interval.map(|v| v as i64),
+            allowed_ips: bincode::Options::serialize(
+                bincode::DefaultOptions::new(),
+                &value.allowed_ips,
+            )
+            .unwrap(),
```

Comment:
Are we really really sure this is safe? I will sleep better at night if the unwrap is removed ;)